### ISSUE TITLE
Stop changing directory

### ISSUE THREAD

**Issue by shulhi** [rescript-lang/vim-rescript#23]
https://github.com/rescript-lang/vim-rescript/commit/d2e9a9023cff41b43359d68b02d25c89f607a1a8 causes undesired effect with other plugins like fzf, nerd tree etc that depends on the current directory.

I am working on a big monorepo that also have other code besides Rescript.

```
- mono-repo
  - not_rescript_project_A
  - not_rescript_project_B
  - rescript_project
    - bs.config.js
    - package.json
  - ....
  - not_rescript_project_Z
  - .git
```

Then,
```
cd mono_repo/rescript_project
vim .
```

I expect `:pwd` to print `mono_repo/rescript_project` but since the plugin changes the directory, it prints `mono_repo`.

Trying to use fzf is now impossible because it tries to search everything under `mono_repo` instead of just `mono_repo/rescript_project`. I can fix it by doing `:cd rescript_project` but still that is troublesome. Is there a way to disable this?

**Comment by ryyppy** [comment]
This is a side-effect I am not entirely sure how to solve, since we use vim's `lcd` command to make sure we are executing our build in the right directory. (see [this line](https://github.com/rescript-lang/vim-rescript/blob/master/autoload/rescript.vim#L51))

It seems to only trigger the NERDtree folder change on the first build, so if I adjust my NERDTree root directory after that, it seems to stay in place.

**Comment by amiralies** [comment]
This is annoying, Can we cd back to the place which editor was started initially after executing build command?

**Comment by ryyppy** [comment]
I don't understand why lcd even triggers that behavior, since it is supposed to change the working directory of the current window, not of the whole editor.

Not sure if switching back and forth between working directories is ideal, but if anyone proposes a clean solution, i am happy to merge.

**Comment by haruki-m** [comment]
The only guaranteed way this is happening for me is that it always changes the directory if the directory you're trying to open doesn't contain the `.git` folder and if the directory it's being changed to contains the `.git` folder. If the current directory contains the `.git` folder, `nvim .` works as expected. My current workaround is just to `mkdir .git` on the folder you want `nvim .` to work on (this directory won't be pushed to the root git repo since the `.git` folder you created is empty).

**Comment by haruki-m** [comment]
What I also found is that it `nvim .` changes directory from the current directory to my home directory (ancestor of current directory) even though the home directory doesn't contain the `.git` folder. My home directory has other hidden folders, so I tested by creating hidden folders with arbitrary names (not `.git`) in my current directory, but I wasn't able to make it work.

**Comment by shulhi** [comment]
> My current workaround is just to `mkdir .git` on the folder you want `nvim .` to work on (this directory won't be pushed to the root git repo since the `.git` folder you created is empty).

That is unnecessary. You can do `:cd your/directory`

**Comment by haruki-m** [comment]
> > My current workaround is just to `mkdir .git` on the folder you want `nvim .` to work on (this directory won't be pushed to the root git repo since the `.git` folder you created is empty).
> 
> That is unnecessary. You can do `:cd your/directory`

I believe you mentioned that doing so every time is troublesome. `mkdir .git`, although creates an unnecessary folder, removes the need to run `:cd you/directory` every time you launch vim. I guess you could put it in your vimrc or something, but ¯\_(ツ)_/¯

**Comment by ryyppy** [comment]
@shulhi @haruki-m Potential fix in #32 

Can you install the changes as denoted by the PR and see if it fixes your issues?

**Comment by shulhi** [comment]
@ryyppy that seems to work! :partying_face:

**Comment by ryyppy** [comment]
okay cool, will cut a release soon then.

**Comment by ryyppy** [comment]
Fixed in #32
